### PR TITLE
added support for assets and templates #3765

### LINKED ISSUE #3765: sechub-api.sh: add api calls for assets and templates
## Situation
The server's api has been extended by calls for managing
- assets
- templates

See https://mercedes-benz.github.io/sechub/latest/sechub-restapi.html

## ToDo
Support above calls in `sechub-api.sh` helper script.

### PR DESCRIPTION
**Pull request by sven-dmlr**
- asset api calls added
- template api calls added
- misc improvements:
  - print HTTP return codes instead of nothing
  - tar and binary upload simplified and fixed
  - check mandatory executables are installed
- code cleanup:
  - lowercase local variables
  - usage output improved
  - 2 blank lines between function declarations
- closes #3765

### CODE REVIEW

File: sechub-developertools/scripts/sechub-api.sh [hunk 1]
```diff
@@ -91,14 +103,26 @@ user_signup_decline <new user-id> - Decline registration of <new user-id>
...
+    if [ ! -x "$exe_path" ] ; then
+        echo "FATAL: Mandatory executable \"$executable\" not found in PATH. Please install."
+        exit 1
+    fi
+}
+
+
 function curl_with_sechub_auth {
   # Don't reveal secrets in the curl process
   printf "user = $SECHUB_USERID:$SECHUB_APITOKEN\n" | curl --config - $CURL_PARAMS "$@"
```
Comment (lorriborri) [inline sechub-developertools/scripts/sechub-api.sh:122]:
```suggestion
  printf "user = $SECHUB_USERID:$SECHUB_APITOKEN\n" | curl --config - "$CURL_PARAMS" "$@"
```

File: sechub-developertools/scripts/sechub-api.sh [hunk 2]
```diff
@@ -268,6 +293,57 @@ function sechub_alive_check {
 }
 
 
+function sechub_asset_file_create {
+  local asset_id="$1"
+  local asset_file="$2"
+  local asset_file_sha256=`sha256sum "$asset_file" | cut -d ' ' -f 1`
```
Comment (lorriborri) [inline sechub-developertools/scripts/sechub-api.sh:299]:
```suggestion
  local asset_file_sha256=$(sha256sum "$asset_file" | cut -d ' ' -f 1)
```

File: sechub-developertools/scripts/sechub-api.sh [hunk 3]
```diff
@@ -377,57 +454,48 @@ function sechub_job_status {
...
 
-  if [[ ! -f "$ZIP_FILE" ]] ; then
-    echo "File \"$ZIP_FILE\" does not exist."
+  if [[ ! -f "$zip_file" ]] ; then
+    echo "File \"$zip_file\" does not exist."
     exit 1
   fi
 
-  local checkSum=$(sha256sum $ZIP_FILE | cut --delimiter=' ' --fields=1)
+  local checkSum=$(sha256sum $zip_file | cut --delimiter=' ' --fields=1)
```
Comment (lorriborri) [inline sechub-developertools/scripts/sechub-api.sh:468]:
```suggestion
  local checkSum=$(sha256sum "$zip_file"| cut --delimiter=' ' --fields=1)
```

File: sechub-developertools/scripts/sechub-api.sh [hunk 4]
```diff
@@ -377,57 +454,48 @@ function sechub_job_status {
...
-  if [[ ! -f "$TAR_FILE" ]] ; then
-    echo "File \"$TAR_FILE\" does not exist."
+  if [[ ! -f "$tar_file" ]] ; then
+    echo "File \"$tar_file\" does not exist."
     exit 1
   fi
 
-  local checkSum=$(sha256sum $TAR_FILE | cut --delimiter=' ' --fields=1)
-  local fileSize=$(ls -l "$TAR_FILE" | cut --delimiter=' ' --fields 5)
+  local checkSum=$(sha256sum $tar_file | cut --delimiter=' ' --fields=1)
```
Comment (lorriborri) [inline sechub-developertools/scripts/sechub-api.sh:487]:
```suggestion
  local checkSum=$(sha256sum "$tar_file" | cut --delimiter=' ' --fields=1)
```

**Review by lorriborri — APPROVED** [review]
Well done! 🎄🎅🏽
I found some minor improvements, but they are optional to fix.

File: sechub-developertools/scripts/sechub-api.sh (hunk 3 above)
Comment (sven-dmlr) [reply sechub-developertools/scripts/sechub-api.sh:468]: changed.
Also for "$tar_file".

File: sechub-developertools/scripts/sechub-api.sh (hunk 1 above)
Comment (sven-dmlr) [reply sechub-developertools/scripts/sechub-api.sh:122]: The `"` will break the curl command.
Leaving them as they are.

File: sechub-developertools/scripts/sechub-api.sh (hunk 4 above)
Comment (sven-dmlr) [reply sechub-developertools/scripts/sechub-api.sh:487]: changed

File: sechub-developertools/scripts/sechub-api.sh (hunk 2 above)
Comment (sven-dmlr) [reply sechub-developertools/scripts/sechub-api.sh:299]: The two forms are equivalent.
Leaving them as they are.

**Pull request merged**